### PR TITLE
Fix preciseExp10

### PR DESCRIPTION
**Pull request by amosbird**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

https://wandbox.org/permlink/2qhQc2jXepz8mbDj  illustrates why it's needed.

For changelog. Remove if this is non-significant change.

Category (leave one):
- Bug Fix

### CODE REVIEW

File: libs/libcommon/src/preciseExp10.c [hunk 1]
```diff
@@ -174,34 +174,52 @@ obstacle to adoption, that text has been removed.
...
+           1e+238, 1e+239, 1e+240, 1e+241, 1e+242, 1e+243, 1e+244, 1e+245, 1e+246, 1e+247, 1e+248, 1e+249, 1e+250, 1e+251, 1e+252, 1e+253, 1e+254,
+           1e+255, 1e+256, 1e+257, 1e+258, 1e+259, 1e+260, 1e+261, 1e+262, 1e+263, 1e+264, 1e+265, 1e+266, 1e+267, 1e+268, 1e+269, 1e+270, 1e+271,
+           1e+272, 1e+273, 1e+274, 1e+275, 1e+276, 1e+277, 1e+278, 1e+279, 1e+280, 1e+281, 1e+282, 1e+283, 1e+284, 1e+285, 1e+286, 1e+287, 1e+288,
+           1e+289, 1e+290, 1e+291, 1e+292, 1e+293, 1e+294, 1e+295, 1e+296, 1e+297, 1e+298, 1e+299, 1e+300, 1e+301, 1e+302, 1e+303, 1e+304, 1e+305,
+           1e+306, 1e+307, 1e+308};
 
     double n, y = modf(x, &n);
-    if (n > 308)
-        return x > 0 ? INFINITY : -INFINITY;
+    if (n > 308) return INFINITY;
```
Comment (akuzm) [inline libs/libcommon/src/preciseExp10.c]: github doesn't let me comment there, but below there is a special code that does something like `fabs(n) < 15`, but in a way that's resistant to NaN values:
```
    if ((u.i >> 52 & 0x7ff) < 0x3ff + 4)
...
 ```

We should decide whether we support NaN values or not, and modify either of the comparisons accordingly.

**Comment by akuzm** [comment]
> https://wandbox.org/permlink/2qhQc2jXepz8mbDj illustrates why it's needed.

Does not compile for me:
```
prog.cc:6:10: fatal error: double-conversion/double-conversion.h: No such file or directory
    6 | #include <double-conversion/double-conversion.h>
```

Is the idea that musl exp10 gives wrong result for some numbers, so we have to use an extended version? What exactly numbers are these?

**Comment by amosbird** [comment]
> Does not compile for me:

Yeah, you need to grab double-conversion  ( can just use the contrib one)

> What exactly numbers are these?

Don't remember now, but it's easy to reproduce with the given code snippet.

File: libs/libcommon/src/preciseExp10.c (hunk 1 above)
Comment (amosbird) [reply libs/libcommon/src/preciseExp10.c]: Passing NAN into preciseExp10 returns NAN currently. Otherwish there is no way to generate NAN values in this function.

**Pull request merged**